codility_training/lessons.lesson10.PrimeAndCompositeNumbers.MinPerimeterRectangle/khw970421.js

```diff
@@ -0,0 +1,23 @@
+//정확성 효율성 100
+function solution(N)
+{
+    let count=0;
+    let num=[];
+
+    let q=Math.sqrt(N);
+    if(Number.isInteger(q))  //약수가 홀수개일때 가장 작은 값은 루트한 값을 4배한것
+        return 4*q;
+
+    for(let i=1; i<=Math.sqrt(N); i++) //약수가 짝수개일때 루트값 전까지 약수를 구한다
+    {
+        if(N % i == 0)
+        {
+            count++;
+            num.push(i);
+        }
+    }
+
+    const l=num.length;
+    let k = N/num[l-1];      //가장 마지막 값은 N의 약수의 중간값이기 때문에 그다음 값 k를구한다. ex) 8=40/5
+    return 2*(k+num[l-1]);   //두개를 더하고 곱한값을 return
+}
```

Comment:
```suggestion
function solution(N)
{
    const num=[];
    const q=Math.sqrt(N);
    if(Number.isInteger(q))  //약수가 홀수개일때 가장 작은 값은 루트한 값을 4배한것
        return 4 * q;

    for(let i=1; i<=q; i++) //약수가 짝수개일때 루트값 전까지 약수를 구한다
    {
        if(N % i == 0)
        {
            num.push(i);
        }
    }

    const last = num.pop();
    const k = N / last;      //가장 마지막 값은 N의 약수의 중간값이기 때문에 그다음 값 k를구한다. ex) 8=40/5
    return 2 * (k + last);   //두개를 더하고 곱한값을 return
}
```
- `count`는 사용되지 않음
- `let` `const` 구분해서 사용할것
- 중복으로 나오는 값은 변수로 뺄것
- q를 선언해놓고 뒤에서 다시 풀어서 사용함